sdk/core/Azure.Core/src/AccessToken.cs

```diff
@@ -32,6 +33,22 @@ public AccessToken(string accessToken, DateTimeOffset expiresOn, DateTimeOffset?
             Token = accessToken;
             ExpiresOn = expiresOn;
             RefreshOn = refreshOn;
+            TokenType = "Bearer";
+        }
+
+        /// <summary>
+        /// Creates a new instance of <see cref="AccessToken"/> using the provided <paramref name="accessToken"/> and <paramref name="expiresOn"/>.
+        /// </summary>
+        /// <param name="accessToken">The access token value.</param>
+        /// <param name="expiresOn">The access token expiry date.</param>
+        /// <param name="refreshOn">Specifies the time when the cached token should be proactively refreshed.</param>
+        /// <param name="tokenType">The access token type.</param>
+        public AccessToken(string accessToken, DateTimeOffset expiresOn, DateTimeOffset? refreshOn, string tokenType)
+        {
+            Token = accessToken;
+            ExpiresOn = expiresOn;
+            RefreshOn = refreshOn;
+            TokenType = tokenType;
```

Comment:
Should there be any validation on an arbitrary string?  Would it have any security implications if someone passed something malicious for this?

Reply:
Generally this type is not constructed by users except, perhaps, in our own tests. It is mostly used by implementations of `TokenCredential`. I don't believe there is any validation that we need to do here.

Reply:
I believe you that if we create an instance, it will not have problems.  Are we sure that there is no way for an entity we don't control to create an instance?  It looks like Azure.Core has a [DelegatedTokenCredential](https://learn.microsoft.com/en-us/dotnet/api/azure.core.delegatedtokencredential?view=azure-dotnet) type that lets someone return an AccessToken.  How is this used?  Would all implementations of types that accept an AccessToken be robust to throwing some exception if they got a token instance with an unrecognized value here?

Reply:
I would categorize this as validating service inputs, which is against our guidelines. The recipient of an AccessToken's Token value is an Azure resource, so it is up to them to properly validate them. Values are considered opaque to clients as they can be encrypted in some scenarios. Note that there is nothing preventing users from providing a bogus token value to REST APIs either.

Reply:
Sounds good.